fenix/app/src/test/java/org/mozilla/fenix/extension/WebExtensionPromptFeatureTest.kt

```diff
@@ -0,0 +1,202 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.extension
+
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.runs
+import io.mockk.spyk
+import io.mockk.verify
+import mozilla.components.browser.state.action.WebExtensionAction.UpdatePromptRequestWebExtensionAction
+import mozilla.components.browser.state.state.extension.WebExtensionPromptRequest
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.concept.engine.webextension.WebExtensionInstallException
+import mozilla.components.support.ktx.android.content.appVersionName
+import mozilla.components.support.test.ext.joinBlocking
+import mozilla.components.support.test.robolectric.testContext
+import mozilla.components.support.test.rule.MainCoroutineRule
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.fenix.R
+import org.mozilla.fenix.helpers.FenixRobolectricTestRunner
+
+@RunWith(FenixRobolectricTestRunner::class)
+class WebExtensionPromptFeatureTest {
+
+    private lateinit var webExtensionPromptFeature: WebExtensionPromptFeature
+    private lateinit var store: BrowserStore
+
+    @get:Rule
+    val coroutinesTestRule = MainCoroutineRule()
+
+    @Before
+    fun setup() {
+        store = BrowserStore()
+        webExtensionPromptFeature = spyk(
+            WebExtensionPromptFeature(
+                store = store,
+                context = testContext,
+                fragmentManager = mockk(relaxed = true),
+            ),
+        )
+    }
+
+    @Test
+    fun `WHEN InstallationFailed is dispatched THEN handleInstallationFailedRequest`() {
```

Comment:
```suggestion
    fun `WHEN InstallationFailed is dispatched THEN handleInstallationFailedRequest is called`() {
```

maybe? not sure how we should read these BDD-style test names.